servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HeaderUtils.java

```diff
@@ -177,7 +177,11 @@ public boolean mapTerminal() {
                                                       final BiIntConsumer<HttpHeaders> contentLengthUpdater) {
         if (messageBody == empty() || (isPayloadEmpty(metadata) && !mayHaveTrailers(metadata))) {
             contentLengthUpdater.apply(0, metadata.headers());
-            return from(metadata, EmptyHttpHeaders.INSTANCE).concat(messageBody.ignoreElements());
+            return messageBody == empty() ?
```

Comment:
nit: does it make sense to cache this?

```java
final boolean messageBodyEmpty = messageBody == empty();
if (messageBodyEmpty || ..) {
  return messageBodyEmpty ? ...;
}
```

Reply:
because it's a comparison of a reference address, not sure cache will be helpful

Reply:
also hope that compiler is smart enough :)